crates/ide-assists/src/handlers/extract_function.rs

```diff
@@ -485,6 +503,20 @@ impl FunctionBody {
         }
     }
 
+    fn node(&self) -> &SyntaxNode {
+        match self {
+            FunctionBody::Expr(e) => e.syntax(),
+            FunctionBody::Span { parent, .. } => parent.syntax(),
+        }
+    }
+
+    fn extracted_from_trait_impl(&self) -> bool {
+        match self.node().ancestors().find_map(ast::Impl::cast) {
```

Comment:
I think this won't work if the function comes from a macro. But that might be okay for now.